ivadomed/loader/adaptative.py

```diff
@@ -175,122 +175,47 @@ class Bids_to_hdf5:
     """Converts a BIDS dataset to a HDF5 file.
 
     Args:
-        root_dir (str): Path to the BIDS dataset.
-        subject_lst (list): Subject names list.
-        target_suffix (list): List of suffixes for target masks.
-        roi_params (dict): Dictionary containing parameters related to ROI image processing.
-        contrast_lst (list): List of the contrasts.
+        dataset (BidsDataset): Bids dataset.
         hdf5_name (str): Path and name of the hdf5 file.
-        contrast_balance (dict): Dictionary controlling image contrasts balance.
         slice_axis (int): Indicates the axis used to extract slices: "axial": 2, "sagittal": 0, "coronal": 1.
         metadata_choice (str): Choice between "mri_params", "contrasts", None or False, related to FiLM.
         slice_filter_fn (SliceFilter): Class that filters slices according to their content.
-        transform (Compose): Transformations.
-        object_detection_params (dict): Object detection parameters.
 
     Attributes:
-        bids_ds (BIDS): BIDS dataset.
         dt (dtype): hdf5 special dtype.
         hdf5_file (hdf5): hdf5 file containing dataset information.
         filename_pairs (list): A list of tuples in the format (input filename list containing all modalities,ground \
             truth filename, ROI filename, metadata).
-        metadata (dict): Dictionary containing metadata of input and gt.
         prepro_transforms (Compose): Transforms to be applied before training.
         transform (Compose): Transforms to be applied during training.
         has_bounding_box (bool): True if all metadata contains bounding box coordinates, else False.
         slice_axis (int): Indicates the axis used to extract slices: "axial": 2, "sagittal": 0, "coronal": 1.
         slice_filter_fn (SliceFilter): Object that filters slices according to their content.
     """
 
-    def __init__(self, root_dir, subject_lst, target_suffix, contrast_lst, hdf5_name, contrast_balance=None,
```

Comment:
Remove code duplication with [BidsDataset](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/loader/loader.py#L750). Most of the code was copied from there with minor additions.